server/service/src/invoice/outbound_return/update/mod.rs

```diff
@@ -0,0 +1,331 @@
+use repository::{
+    Invoice, InvoiceRowRepository, InvoiceRowStatus, RepositoryError, StockLineRowRepository,
+};
+
+use crate::{
+    activity_log::{activity_log_entry, log_type_from_invoice_status},
+    invoice::get_invoice,
+    service_provider::ServiceContext,
+};
+
+pub mod generate;
+pub mod validate;
+use generate::generate;
+use validate::validate;
+
+use self::generate::GenerateResult;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum UpdateOutboundReturnStatus {
+    Picked,
+    Shipped,
```

Comment:
Have removed `Allocated` as a status here compared to outbound shipment, as when we generate the return lines, we always include a stock line id (i.e. we won't have placeholder lines with returns) - just want to double check my understanding is correct there?